docs/contracts/v4/concepts/02-pool-manager-and-pool.mdx

```diff
@@ -0,0 +1,175 @@
+
+1. **Check the Fee**:
+- If the fee for using the pool is too high, then revert error called `FeeTooLarge`.
+
+2. **Check the Tick Spacing**:
+- "Ticks" are like markers in the pool. There's a space between these markers.
+- If the space between the markers is too large, then revert with error `TickSpacingTooLarge`.
+- If it's too small, then revert with error `TickSpacingTooSmall`.
+
+3. **Ensure the Digital Currencies are in Order**:
+- Keys should be sorted numerically
+- If they are in the wrong order, then revert with error `CurrenciesInitializedOutOfOrder`.
+
+4. **Check the Hooks**:
+
+5. **Set Up the Pool**:
+- Create a unique `id` for the pool.
+- Actually set up the pool with the given starting price and fees.
+
+6. **After Setup Actions**:
+- Some hooks might have actions to do after the pool is set up. If they don't respond correctly, revert with `InvalidHookResponse`.
+
+7. **Announce the Pool**:
+- Finally, announce to everyone that a new pool has been created, sharing all its details.
+
+# PoolKey
+
+The `PoolKey` is a structure that uniquely identifies a liquidity pool by storing essential details like the two
+currencies involved (sorted numerically), the swap fee, tick spacing, and hooks (extra functionalities) of the pool.
+
+It acts as a unique identifier, ensuring that each pool can be precisely specified and accessed within the code.
+
+```solidity
+/// @notice Returns the key for identifying a pool
+struct PoolKey {
+    /// @notice The lower currency of the pool, sorted numerically
+    Currency currency0;
+    /// @notice The higher currency of the pool, sorted numerically
+    Currency currency1;
+    /// @notice The pool swap fee, capped at 1_000_000. The upper 4 bits determine if the hook sets any fees.
+    uint24 fee;
+    /// @notice Ticks that involve positions must be a multiple of tick spacing
+    int24 tickSpacing;
+    /// @notice The hooks of the pool
+    IHooks hooks;
+}
+```
+
+Since we create and pass the `PoolKey` to the `initialize` function, and as part of the PoolKey we pass the `hooks` we
+want to use for the pool.
+
+We can use the `hooks` to customize the pool to our liking.
+
+# Initialization Code
+
+```solidity
+
+contract TestCounter {
+   Currency internal currency1;
+   Currency internal currency0;
+   uint160 constant SQRT_RATIO_10_1 = 250541448375047931186413801569;
+   TestERC20 token0;
+   TestERC20 token1;
+   PoolManager manager;
+   PoolKey key;
+   PoolId id;
+
+   bytes constant ZERO_BYTES = new bytes(0);
+
+   function setup(){
+
+      MockERC20 tokenA = new MockERC20("TestA", "A", 18, 1000 ether);
+      MockERC20 tokenB = new MockERC20("TestB", "B", 18, 1000 ether);
+
+      (currency0, currency1) = SortTokens.sort(tokenA, tokenB);
+
+      manager = new PoolManager(500000);
+
+
+      key = PoolKey(currency0, currency1, 3000, 60, limitOrder);
+      id = key.toId();
+      manager.initialize(key, SQRT_RATIO_1_1, ZERO_BYTES);
+
+   }
+}
+```
```

Comment:
stale now, initialization requires the lock pattern. See https://www.v4-by-example.org/initialize for snippets

In the future, i think initialization might live on the PositionManager periphery router (doesnt exist yet)